File: src/mrtjp/projectred/illumination/IlluminationRecipes.java
package mrtjp.projectred.illumination;

import mrtjp.projectred.ProjectRedIllumination;
import mrtjp.projectred.core.ItemPart.EnumPart;
import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import cpw.mods.fml.common.registry.GameRegistry;

public class IlluminationRecipes
{
    public static void initRecipes()
    {
        initLightingRecipes();
    }

    private static void initLightingRecipes()
    {
        /** Lamps **/
        for (int i = 0; i < 16; i++)
        {
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.blockLamp(), 1, i),  // Regular
                    "gIg",
                    "gIg",
                    "gtg",
                    'g', Block.thinGlass,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    't', Item.redstone
                    );
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.blockLamp(), 1, i+16), // Inverted
                    "gIg",
                    "gIg",
                    "gtg",
                    'g', Block.thinGlass,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    't', Block.torchRedstoneActive
                    );
        }

        /** Lanterns **/
        for (int i = 0; i < 16; i++)
        {
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.itemPartLantern(), 1, i),
                    "PNP",
                    "GIG",
                    "PRP",
                    'P', EnumPart.PLATE.getItemStack(),
                    'N', Item.goldNugget,
                    'G', Block.thinGlass,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    'R', Item.redstone
                    );
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.itemPartInvLantern(), 1, i),
                    "PNP",
                    "GIG",
                    "PRP",
                    'P', EnumPart.PLATE.getItemStack(),
                    'N', Item.goldNugget,
                    'G', Block.thinGlass,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    'R', Block.torchRedstoneActive
                    );
        }

        /** Buttons **/
        for (int i = 0; i < 16; i++)
            GameRegistry.addShapelessRecipe(
                    new ItemStack(ProjectRedIllumination.itemPartIllumarButton(), 1, i),
                    Block.stoneButton,
                    EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    EnumPart.ILLUMAR_PARTS[i].getItemStack()
                    );

        /** Cage Lamps **/
        for (int i = 0; i < 16; i++)
        {
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.itemPartCageLamp(), 1, i),
                    "CCC",
                    "CIC",
                    "NPN",
                    'C', Block.fenceIron,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    'N', Item.goldNugget,
                    'P', EnumPart.CONDUCTIVEPLATE.getItemStack()
                    );
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.itemPartInvCageLamp(), 1, i),
                    "CCC",
                    "CIC",
                    "NPN",
                    'C', Block.fenceIron,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    'N', Item.goldNugget,
                    'P', EnumPart.CATHODE.getItemStack()
                    );
        }

        /** Fixtures **/
        for (int i = 0; i < 16; i++)
        {
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.itemPartFixture(), 1, i),
                    "ggg",
                    "gIg",
                    "pPp",
                    'g', Block.thinGlass,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    'p', EnumPart.PLATE.getItemStack(),
                    'P', EnumPart.CONDUCTIVEPLATE.getItemStack()
                    );
            GameRegistry.addRecipe(new ItemStack(ProjectRedIllumination.itemPartInvFixture(), 1, i),
                    "ggg",
                    "gIg",
                    "pPp",
                    'g', Block.thinGlass,
                    'I', EnumPart.ILLUMAR_PARTS[i].getItemStack(),
                    'p', EnumPart.PLATE.getItemStack(),
                    'P', EnumPart.CATHODE.getItemStack()
                    );
        }
    }
}

File: src/mrtjp/projectred/illumination/IlluminationRecipes.scala
package mrtjp.projectred.illumination

import java.lang.{Character => JC}

import mrtjp.projectred.ProjectRedIllumination
import mrtjp.projectred.core.PartDefs
import net.minecraft.init.Blocks
import net.minecraft.item.ItemStack
import net.minecraftforge.fml.common.registry.GameRegistry
import net.minecraftforge.oredict.ShapedOreRecipe

object IlluminationRecipes
{
    def initRecipes()
    {
        initLighting()
    }

    private def initLighting()
    {
        /** Lamps **/
        for (i <- 0 until 16)
        {
            GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ProjectRedIllumination.blockLamp, 1, i),
                "gIg",
                "gIg",
                "gtg",
                'g':JC, "paneGlassColorless",
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                't':JC, "dustRedstone"
            ))

            GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ProjectRedIllumination.blockLamp, 1, i+16),
                "gIg",
                "gIg",
                "gtg",
                'g':JC, "paneGlassColorless",
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                't':JC, Blocks.REDSTONE_TORCH
            ))
        }

        /** Lanterns **/
        for (i <- 0 until 16)
        {
            GameRegistry.addRecipe(new ShapedOreRecipe(LightFactoryLantern.makeStack(i),
                "PNP",
                "GIG",
                "PRP",
                'P':JC, PartDefs.PLATE.makeStack,
                'N':JC, "nuggetGold",
                'G':JC, "paneGlassColorless",
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'R':JC, "dustRedstone"
            ))
            GameRegistry.addRecipe(new ShapedOreRecipe(LightFactoryLantern.makeInvStack(i),
                "PNP",
                "GIG",
                "PRP",
                'P':JC, PartDefs.PLATE.makeStack,
                'N':JC, "nuggetGold",
                'G':JC, "paneGlassColorless",
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'R':JC, Blocks.REDSTONE_TORCH
            ))
        }

        /** Buttons **/
        for (i <- 0 until 16)
        {
            GameRegistry.addShapelessRecipe(new ItemStack(ProjectRedIllumination.itemPartIllumarButton, 1, i),
                Blocks.STONE_BUTTON,
                PartDefs.ILLUMARS.toSeq(i).makeStack,
                PartDefs.ILLUMARS.toSeq(i).makeStack
            )
            GameRegistry.addShapelessRecipe(new ItemStack(ProjectRedIllumination.itemPartIllumarFButton, 1, i),
                new ItemStack(ProjectRedIllumination.itemPartIllumarButton, 1, i),
                Blocks.REDSTONE_TORCH
            )
        }

        /** Fallout Lights **/
        for (i <- 0 until 16)
        {
            GameRegistry.addRecipe(new ShapedOreRecipe(LightFactoryFallout.makeStack(i),
                "CCC", "CIC", "NPN",
                'C':JC, Blocks.IRON_BARS,
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'N':JC, "nuggetGold",
                'P':JC, PartDefs.CONDUCTIVEPLATE.makeStack
            ))

            GameRegistry.addRecipe(new ShapedOreRecipe(LightFactoryFallout.makeInvStack(i),
                "CCC", "CIC", "NPN",
                'C':JC, Blocks.IRON_BARS,
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'N':JC, "nuggetGold",
                'P':JC, PartDefs.CATHODE.makeStack
            ))
        }

        /** Cage Lamps **/
        for (i <- 0 until 16)
        {
            GameRegistry.addRecipe(LightFactoryCage.makeStack(i),
                " C ", "CIC", "pPp",
                'C':JC, Blocks.IRON_BARS,
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'p':JC, PartDefs.PLATE.makeStack,
                'P':JC, PartDefs.CONDUCTIVEPLATE.makeStack
            )

            GameRegistry.addRecipe(LightFactoryCage.makeInvStack(i),
                " C ", "CIC", "pPp",
                'C':JC, Blocks.IRON_BARS,
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'p':JC, PartDefs.PLATE.makeStack,
                'P':JC, PartDefs.CATHODE.makeStack
            )
        }

        /** Fixtures **/
        for (i <- 0 until 16)
        {
            GameRegistry.addRecipe(new ShapedOreRecipe(LightFactoryFixture.makeStack(i),
                "ggg", "gIg", "pPp",
                'g':JC, "paneGlassColorless",
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'p':JC, PartDefs.PLATE.makeStack,
                'P':JC, PartDefs.CONDUCTIVEPLATE.makeStack
            ))
            GameRegistry.addRecipe(new ShapedOreRecipe(LightFactoryFixture.makeInvStack(i),
                "ggg", "gIg", "pPp",
                'g':JC, "paneGlassColorless",
                'I':JC, PartDefs.ILLUMARS.toSeq(i).makeStack,
                'p':JC, PartDefs.PLATE.makeStack,
                'P':JC, PartDefs.CATHODE.makeStack
            ))
        }
    }
}
